sdk/core/System.ClientModel/src/Convenience/ApiKeyCredential.cs

```diff
@@ -66,6 +67,38 @@ public void Update(string key)
         Volatile.Write(ref _key, key);
     }
 
-    /// <summary> Converts a string to an <see cref="ApiKeyCredential"/>. </summary>
-    public static implicit operator ApiKeyCredential(string key) => new(key);
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ApiKeyCredential"/> class
+    /// from the value stored in the specified environment variable.
+    /// </summary>
+    /// <param name="variable">
+    /// The name of the environment variable where the API key is stored.
+    /// </param>
+    /// <returns></returns>
+    /// <exception cref="System.ArgumentNullException">
+    /// Thrown when the <paramref name="variable"/> is null.
+    /// </exception>
+    /// <exception cref="System.ArgumentException">
+    /// Thrown when the <paramref name="variable"/> is empty.
+    /// </exception>
+    /// <exception cref="ArgumentException">
+    /// Thrown when the environment variable is not found.
+    /// </exception>
+    /// <exception cref="SecurityException">
+    /// Thrown when the caller does not have the required permission to perform
+    /// this operation.
+    /// </exception>
+    public static ApiKeyCredential FromEnvironmentVariable(string variable)
```

Comment:
Do we need this?   It seems that if someone is savvy enough to set an environment variable, then knowing how to read it back to pass to the standard constructor is a reasonable expectation.  I'm not sure this buys us much, especially with the discovery tax.

```csharp
var cred = ApiKeyCredential
    .FromEnvironmentVariable("SHARED_KEY");
```

versus

```csharp
var cred = new ApiKeyCredential(
    Environment.GetEnvironmentVariable("SHARED_KEY"));
```

Reply:
The idea is that, since this appears in basically every sample, this addition makes them look a little better, which helps give a good first impression. But I personally don't feel strongly either way.

I removed the factory method and repurposed this PR to simply removing the implicit cast from string to ApiKeyCredential.